src/sync.cpp

```diff
@@ -436,11 +436,11 @@ bool assignFilesystemIds(Sync& sync, MegaApp& app, FileSystemAccess& fsaccess, h
     // Ensures that unmatched nodes (local nodes that don't have a fingerprint that's
     // the same as a file on disk) have invalid IDs.
     size_t invalidatedCount = 0;
-    invalidateFilesystemIds(fsidnodes, sync.localroot, invalidatedCount);
+    invalidateFilesystemIds(fsidnodes, *sync.localroot, invalidatedCount);
```

Comment:
Why not with `sync->localroot`?? here and 4 lines below too

Reply:
`localroot` is a pointer now, so that we can destroy its object separately in ~Sync (with the db transaction around that).  But the function called here takes a LocalNode& so we need to dereference it now.  cheers

Reply:
Oh, right. I thought the pointer here was the `sync` and that `invalidateFilesystemIds()` too a pointer to `LocalNode`. Nothing else to say :)